docs/30-data-valuation.rst

```diff
@@ -23,118 +23,380 @@ interest, but a function of three factors:
+  to the values. Nevertheless, in practice it is only the ranking of samples
+  that matters, and this tends to be accurate (wrt. to the true ranking) despite
+  inaccurate values.
+
+  pyDVL offers a dedicated :func:`function composition
+  <pydvl.utils.types.compose_score>` for scorer functions which can be used to
+  squash a score. The following is defined in module :mod:`~pydvl.utils.numeric`:
+
+  .. code-block:: python
+
+     def sigmoid(x: float) -> float:
+         return float(1 / (1 + np.exp(-x)))
+
+     squashed_r2 = compose_score("r2", sigmoid, "squashed r2")
+
+     squashed_variance = compose_score(
+         "explained_variance", sigmoid, "squashed explained variance"
+     )
+
+  These squashed scores can prove useful in regression problems, but they can
+  also introduce issues in the low-value regime.
+
+* **High variance utility**: Classical applications of game theoretic value
+  concepts operate with deterministic utilities, but in ML we use an evaluation
+  of the model on a validation set as a proxy for the true risk. Even if the
+  utility *is* bounded, if it has high variance then values will also have high
+  variance, as will their Monte Carlo estimates. One workaround in pyDVL is to
+  configure the caching system to allow multiple evaluations of the utility for
+  every index set. A moving average is computed and returned once the standard
+  error is small, see :class:`~pydvl.utils.config.MemcachedConfig`.
+
+  :footcite:t:`wang_data_2022` prove that by relaxing one of the Shapley axioms
+  and considering the general class of semi-values, of which Shapley is an
+  instance, one can prove that a choice of constant weights is the best one can
+  do in a utility-agnostic setting. So-called *Data Banzhaf* is on our to-do
+  list!
+
+* **Data set size**: Computing exact Shapley values is NP-hard, and Monte Carlo
+  approximations can converge slowly. Massive datasets are thus impractical, at
+  least with current techniques. A workaround is to group samples and investigate
+  their value together. In pyDVL you can do this using
+  :class:`~pydvl.utils.dataset.GroupedDataset`. There is a fully worked-out
+  :doc:`example here <examples/shapley_basic_spotify>`. Some algorithms also
+  provide different sampling strategies to reduce the variance, but due to a
+  no-free-lunch-type theorem, no single strategy can be optimal for all
+  utilities.
+
+* **Model size**: Since every evaluation of the utility entails retraining the
+  whole model on a subset of the data, large models require great amounts of
+  computation. But also, they will effortlessly interpolate small to medium
+  datasets, leading to great variance in the evaluation of performance on the
+  dedicated validation set. One mitigation for this problem is cross-validation,
+  but this would incur massive computational cost. As of v.0.3.0 there are no
+  facilities in pyDVL for cross-validating the utility (note that this would
```

Comment:
I think this is currently doable by subclassing the `Utility` and using cross validation in the `_utility` method.

Reply:
Isn't that precisely proof that we don't provide a ready solution? That would be a `CrossValidatedUtility` ready to be used.

Reply:
You're right. The way you phrased it in the docs is truer.
Still, I think that gives off the idea that it is impossible without us implementing it first.
Perhaps we could showcase this possibility in a new notebook in which we could show more of these data valuation issues.

Reply:
Sounds like a good idea!

Reply:
https://github.com/appliedAI-Initiative/pyDVL/issues/198